frontend/include/chpl/framework/ID.h

```diff
@@ -176,6 +176,9 @@ class ID final {
    */
   ID parentSymbolId(Context* context) const;
 
+  /** Returns the parent function ID, or the empty ID if there is none. */
+  ID parentFunctionId(Context* context) const;
```

Comment:
I don't think this belongs as a method on `ID`. It might seem similar to `parentSymbolId` but that is just using `Context*` to create some `UniqueStrings`. I think that something that computes the parent function ID belongs in parsing-queries.cpp. Compare with `idToParentModule`. I don't know that it is or isn't a good idea to implement it with `Scope`s. Conceptually, it's something that exists in the nesting of the uAST so I would put it in parsing-queries.cpp and not use scopes to implement it.